src/builder/pod/volume.rs

```diff
@@ -346,6 +360,29 @@ impl SecretOperatorVolumeSourceBuilder {
     }
 }
 
+/// A [secret format](https://docs.stackable.tech/home/stable/secret-operator/secretclass.html#format) known by secret-operator.
+///
+/// This must either match or be convertible from the corresponding secret class, or provisioning the volume will fail.
+#[derive(Clone)]
+pub enum SecretFormat {
+    /// A TLS certificate formatted as a PEM triple (`ca.crt`, `tls.crt`, `tls.key`) according to Kubernetes conventions.
+    Tls,
+    /// A TLS certificate formatted as a PKCS#12 store.
+    TlsPkcs12,
+    /// A Kerberos keytab.
+    Kerberos,
+}
+
+impl SecretFormat {
+    fn as_str(&self) -> &'static str {
+        match self {
+            SecretFormat::Tls => "tls",
+            SecretFormat::TlsPkcs12 => "tls-pkcs12",
+            SecretFormat::Kerberos => "kerberos",
+        }
+    }
+}
```

Comment:
It might be worth to impl `Display`. This would also get rid of `.as_str().to_string()` above. It might even be possible to use `strum` as we already use it in some other place IIRC.